lib/ProductOpener/APITest.pm

```diff
@@ -166,17 +320,98 @@ For the example cited above this returns: "http://world-fr.openfoodfacts.localho
 
 =cut
 
-sub construct_test_url ($target, $prefix) {
-	my $link = "openfoodfacts.localhost";
-	# my $api_end = "/api/v2/search?";
-	my $api_end = "api/v2";
-
-	if (index($target, $api_end) != -1) {
+sub construct_test_url ($target, $prefix = "world") {
+	my $link = $TEST_MAIN_DOMAIN;
+	# no cgi inside url ? add display.pl
+	if (index($target, "cgi") == -1) {
```

Comment:
I find it odd to use index to see if the url contains cgi (and in fact we should check that it starts with /cgi)

```suggestion
	if ($target =~ /^\/cgi\//) {
```